src/battle_main.c

```diff
@@ -4236,7 +4236,8 @@ u32 GetBattlerTotalSpeedStat(u8 battlerId)
     // player's badge boost
     if (!(gBattleTypeFlags & (BATTLE_TYPE_LINK | BATTLE_TYPE_x2000000 | BATTLE_TYPE_FRONTIER))
         && FlagGet(FLAG_BADGE03_GET)
-        && GetBattlerSide(battlerId) == B_SIDE_PLAYER)
+        && GetBattlerSide(battlerId) == B_SIDE_PLAYER
+        && B_BADGE_BOOST == GEN_3)
```

Comment:
You might as well use `ShouldGetStatBadgeBoost` now that you've defined it.

Reply:
I didn't touch the original parameters that are being checked. Is there a good reason to actually change the `FlagGet` for the reimplemented function besides "just because" or "just to make use of it"? Like, an actual improvement of any kind?

Reply:
Yes, using `ShouldGetStatBadgeBoost` is an improvement. If someone wants to change how `ShouldGetStatBadgeBoost` works, then changing it in the one place should change the behavior for all badge boosts. Currently, they would have to remember to change this area too. So making all the different badge boost checks use the one `ShouldGetStatBadgeBoost` function ends up being more maintainable.